unittests/baseplatformtests.py

```diff
@@ -171,22 +171,22 @@ def _run(self, command, *, workdir=None, override_envvars: T.Optional[T.Mapping[
             env = os.environ.copy()
             env.update(override_envvars)
 
-        p = subprocess.run(command, stdout=subprocess.PIPE,
-                           stderr=subprocess.STDOUT if stderr else subprocess.PIPE,
-                           env=env,
-                           encoding='utf-8',
-                           text=True, cwd=workdir, timeout=60 * 5)
+        proc = subprocess.run(command, stdout=subprocess.PIPE,
+                              stderr=subprocess.STDOUT if stderr else subprocess.PIPE,
+                              env=env,
+                              encoding='utf-8',
+                              text=True, cwd=workdir, timeout=60 * 5)
         print('$', join_args(command))
         print('stdout:')
-        print(p.stdout)
+        print(proc.stdout)
         if not stderr:
             print('stderr:')
-            print(p.stderr)
-        if p.returncode != 0:
-            if 'MESON_SKIP_TEST' in p.stdout:
+            print(proc.stderr)
+        if proc.returncode != 0:
+            if 'MESON_SKIP_TEST' in proc.stdout:
                 raise SkipTest('Project requested skipping.')
-            raise subprocess.CalledProcessError(p.returncode, command, output=p.stdout)
-        return p.stdout
+            raise subprocess.CalledProcessError(proc.returncode, command, output=proc.stdout)
+        return proc.stdout
```

Comment:
Again, split this change out into its own commit.

Reply:
I could do that, but I'd like to better understand why you think that's necessary/preferable.

If a change is fairly incidental, superficial, and is deemed OK, is there any practical harm it does to get rolled in with a more substantial change?

Just because something can be done, does that mean it should?  Any time someone renames a variable or corrects some comment spelling/grammer, are you on the case to get it out into a separate commit? Is that worth the shennanigans involved in cherry-picking/juggling changes of this kind?   In what way is this a practical benefit?

I'm not trying to be objectionable but I do think it best to at least make an honest attempt to discuss, understand, and negotiate where I might initially disagree or not see the value of something.

Reply:
Commits should be self-contained and atomic. Your commit message gives it away that it should be split into two commits.

> Fix 'test_lto_mode' and rename variables that clash with pdb commands.

2 completely unrelated changes.

Reply:
> If a change is fairly incidental, superficial, and is deemed OK, is there any practical harm it does to get rolled in with a more substantial change?

It violates the project coding standards, which is "one commit per logical change" and the two logical changes here are:
-  Fix 'test_lto_mode'
- rename variables that clash with pdb commands.

> Just because something can be done, does that mean it should? Any time someone renames a variable or corrects some comment spelling/grammer, are you on the case to get it out into a separate commit? Is that worth the shennanigans involved in cherry-picking/juggling changes of this kind? In what way is this a practical benefit?

It makes it orders of magnitude easier for:
- project maintainers to review the PR (in particular, a commit that only contains "inert" changes such as variable renaming or comment tweaks can be mentally filtered out and allow reviewers to look at just the active changes)
- future readers of the code to understand the flow of changes 
- `git bisect` to accurately pin down which commit affected some functionality (what if a variable name was typoed in an uncommon branch? If the bisect points to a commit that changes 3 or 4 different things, which change caused the regression?)
- backporting patches to maintenance branches (changes to comments or variable names in maintenance branches increases the likelihood of conflicts, and generally constitutes an unnecessary risk)


which is why the *general* guideline is in place, it's not specific to this PR and it's not even specific to the Meson project. In fact, the git software itself includes the same recommendation in its own guidance: https://git-scm.com/docs/SubmittingPatches#separate-commits


If it helps, I find that it becomes much easier to juggle the changes if I start off by adding changes using `git add -p` to interactively add just part of the change to a commit, and then commit the change and use `git add -p` to add other changes to a second commit. It's also easier to start off with two commits and later squash them, than to start off with one commit and later break them apart.

Reply:
Thanks both of you for taking the time to try to elaborate on this point but I'm still struggling to see the real-world net benefits in this particular case.

I'm still left with the impression that there is a general guideline, which seems reasonable, but which has now become sacrosanct and can be, in a few circumstances, of dubious value, if applied to the extreme.
I hope you can see that neither "2 completely unrelated changes" nor discussing general guidlines and more nebulous "understand the flow of changes" are things that I have any problem with, but neither do they specifically explain concrete examples of problems that arise from failing to split incidental changes like renaming out into a separate change.  They just restate a guideline/doctrine or talking in general terms, avoiding specifics, which I find doesn't help clarify.

To re-iterate: I'm not disputing the general niceness of having separate logical changes, but I would like to understand real world examples of the value of splitting some incidental renaming changes out into a separate change.

Renaming variables must happen a lot.  Of those, I'd imagine a significant proportion would be to simply use better, more meaningful variable names.  Now, I don't think anyone could dispute that incidental, one-off, isolated variable renaming to something marginally better is a change that technically CAN be separated out into a new commit, but a whole separate commit of nothing more than 'Rename: More meaningful variable name', for example, still strikes me as of rather dubious value to go to the effort of reorganising commits for some incidental renaming.
But if that were really the case, then huge amounts of commits for all projects would be bloated with the likes of 'Rename (X) to better name (Y)' ... Only we don't have that, in general, because I think there is a more sensible approach than just blindly following the standards/guidlines(/dogma) to the extreme (i.e. in all possible cases).  All of which means that there must be some pragmatic grey area that prevents the blind, unquestioning enforcement/application of the rule: "If it can conceivably be considered a separate logical change then it must be split out into a separate change".  And what defines that grey area must surely come from an understanding of the practical problems that are fixed/avoided by going to the effort of splitting up changes to the Nth degree.  For me, simple renaming improvements (among other things) has always fallen into this grey area of there being no clear benefit to splitting out as a separate change (as well as quite tangible drawbacks of busying commit history with arguably unnecessary noise, adding sometimes fiddly work with potential to introduce mistakes)... but maybe there are scenarios I'm unaware of where this really does pay off.

To avoid talking in less helpful, general terms, or merely refering to rules/guidelines as if that should be answer enough, in the specific case of incidental variable renaming, I would appreciate it if you could humour me with some real-world scenarios that might better illustrate the merits of going to the effort of splitting them out into separate changes, as well as clarifying if you think there's a pragmatic grey area where it's not worth splitting out spearate changes to Nth degree, and if so, what distinguishes that.

Reply:
A real world example is that people don't like reviewing commits with changes that aren't logically the same. It's a context switch for no logical reason.

Variable name changes typically happen in the course of other changes in the same unit of code.

You are also hurting the chances that anyone gives you a serious review.

Reply:
> a context switch for no logical reason

Thank you. I think this is perhaps the closest we've come to getting something tangible to grasp on to what a possible problem might be (and which is not too broad a generalisation or "Them's the rules; they must be applied to the Nth degree").  I didn't intend for it to be a difficult or trick question.  However, I do think it's quite a stretch of the imagination to think that such a simple, incidental renaming change could realistically give any reviewer/reader any difficulty at all... but I do conceed that perhaps there's a remote chance that, even in this case, it does.

Anyway, on this point, even though I think I still disagree with the problem that this renaming (being lumped in with another change) realistically imposes, and I stand by the view that I think it is a tangible, error prone risk to unpick changes of this type, with drawbacks that outweigh the benefits of splitting out, I've done this change.

> You are also hurting the chances that anyone gives you a serious review.

By doing what?  I'm not refusing to make changes, but I am asking for clear explanation if I don't understand or dispute the value of changes.  I think it's the job of a responsible, diligent reviewee to not merely unquestioningly do as they're asked, but to truly understand the value of suggestions and to also challenge what is being asked if the benefits are unclear. This way, both the reviewee can learn, but also the reviewer has the opportunity to learn, to question the extent to which they understand the value of their practices; to be a better communicator as well as to refine and develop pragmatic and nuanced opinions (or even harden their views) under questioning.  I also hope a good reviewer would take questions at face value (to not read any malicious intent or get upset at being pressed to better explain and justify their view), and be able to discuss, negotiate, learn, change, and be flexible if debate ensues.

Thanks again both of you (@tristan957 and @eli-schwartz ) for taking the time to discuss this.